### DISCUSSION TITLE
Route [] not defined.

### DISCUSSION THREAD

**Discussion by rabol** [creagia/laravel-sign-pad#19] [Q&A] [answered]
Hi

I get the sign-pad (the drawing is a litlle off compared to the mouse pointer, but it works), I do a signature and hit submit.

then I get an error: Route [] not defined.

however the url in the address bar looks like this: ```http://blog.test/creagia/sign-pad?model=App%5CModels%5CMyModel&id=4&token=8a7089a37d3eb4a650728cd689a7125b```

Did I miss something in the installation ?

kind regards
Steen

**Comment by dtorras** [comment, marked as answer]
It looks like your Laravel installation is not loading routes from the package. Have you tried cleaning the routes cache? Another option is that you didn't define the redirect route name. Can you check both?

Not part of this problem, but you can define `method='post'` in the form to avoid passing the parameters on the URL:

```blade
@if (!$myModel->hasBeenSigned())
    <form action="{{ $myModel->getSignatureRoute() }}" method="POST">
        @csrf
        <div style="text-align: center">
            <x-creagia-signature-pad />
        </div>
    </form>
    <script src="{{ asset('vendor/sign-pad/sign-pad.min.js') }}"></script>
@endif
```

**Reply by rabol** [reply]
I did not read all of the read me - I forgot to setup the ```redirect_route_name```
